src/partner-access/zapier-auth.guard.ts

```diff
@@ -0,0 +1,22 @@
+import { CanActivate, ExecutionContext, UnauthorizedException } from '@nestjs/common';
+import { Request } from 'express';
+
+export class ZapierAuthGuard implements CanActivate {
+  canActivate(context: ExecutionContext): boolean {
+    const request = context.switchToHttp().getRequest<Request>();
+
+    const { authorization } = request.headers;
+
+    if (!authorization) {
+      throw new UnauthorizedException('Unauthorized: missing required Basic auth');
+    }
+
+    const zapierToken = authorization.split('Basic ')[1];
+
+    if (zapierToken !== process.env.ZAPIER_TOKEN) {
+      throw new UnauthorizedException('Unauthorized: invalid auth token');
+    }
+
+    return true;
+  }
+}
```

Comment:
we set up a username/password on the zapier webhooks, shall we include a check here for those? I guess storing the username/password as env variables?

Reply:
The username and password aren't sent back in the request header or in the body of the object.